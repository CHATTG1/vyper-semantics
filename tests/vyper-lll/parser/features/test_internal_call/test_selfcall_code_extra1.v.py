@public
def foo(x:num) -> num:
    return x + 1

@public
def bar(x:num):
    a = self.foo(self.foo(x))

