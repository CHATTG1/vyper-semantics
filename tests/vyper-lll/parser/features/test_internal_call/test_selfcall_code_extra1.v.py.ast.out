$%seq ( $return ( 0 , $lll ( $%seq ( $%seq ( $mstore ( 28 , $calldataload ( 0 ) ) , $mstore ( 32 , 1461501637330902918203684832716283019655932542976 ) , $mstore ( 64 , 170141183460469231731687303715884105727 ) , $mstore ( 96 , -170141183460469231731687303715884105728 ) , $mstore ( 128 , 1701411834604692317316873037158841057270000000000 ) , $mstore ( 160 , -1701411834604692317316873037158841057280000000000 ) ) , $if ( $eq ( $mload ( 0 ) , 3650092561 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $%seq ( $mstore ( 0 , $clamp ( $mload ( 96 ) , $add ( $mload ( 320 ) , 1 ) , $mload ( 64 ) ) ) , $return ( 0 , 32 ) ) , $stop ) ) , $if ( $eq ( $mload ( 0 ) , 1091910332 ) , $%seq ( $calldatacopy ( 320 , 4 , 32 ) , $assert ( $iszero ( $callvalue ) ) , $clamp ( $mload ( 96 ) , $calldataload ( $add ( 4 , 0 ) ) , $mload ( 64 ) ) , $mstore ( 672 , $mload ( $%seq ( $assert ( $call ( $gas , $address , 0 , $%seq ( $mstore ( 512 , 3650092561 ) , $mstore ( 544 , $mload ( $%seq ( $assert ( $call ( $gas , $address , 0 , $%seq ( $mstore ( 352 , 3650092561 ) , $mstore ( 384 , $mload ( 320 ) ) , 380 ) , 36 , 480 , 32 ) ) , 480 ) ) ) , 540 ) , 36 , 640 , 32 ) ) , 640 ) ) ) , $stop ) ) ) , 0 ) ) )
